chanfitness/chanfitness.go

```diff
@@ -0,0 +1,226 @@
+// Package chanfitness monitors the behaviour of channels to provide scores
+// which can be factored into the decision to close a channel.
+//
+// The metrics monitored are:
+// Uptime: the total time that the channel's remote peer has been observed as
+// being online by the score store.
+//
+// Lifetime: the total time the channel has been known to the scoring system.
+// Note that this does not equal channel lifetime because score data is not
+// currently persisted.
+//
+// Flap count: average the number of times the channel's remote peer has been
+// observed going offline.
+package chanfitness
+
+import (
+	"sync"
+	"time"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/channelnotifier"
+	"github.com/lightningnetwork/lnd/lnpeer"
+	"github.com/lightningnetwork/lnd/subscribe"
+)
+
+// ScoreStore maintains a set of scores for the node's channels. It is intended
```

Comment:
Instead of the notion of this storing a "score", should we change it to store "events". The score can later be calculated on the basis of the events. We can name this `ChannelEventStore`?